tracing-mock/src/span.rs

```diff
@@ -188,6 +213,100 @@ impl ExpectedSpan {
                 name: Some(name.into()),
                 ..self.metadata
             },
+            ..self
+        }
+    }
+
+    /// Sets the `ID` to expect when matching a span.
+    ///
+    /// The [`ExpectedId`] can be used to differentiate spans that are
+    /// otherwise identical. An [`ExpectedId`] needs to be attached to
+    /// an `ExpectedSpan` or [`NewSpan`] which is passed to
+    /// [`MockCollector::new_span`]. The same [`ExpectedId`] can then
+    /// be used to match the exact same span when passed to
+    /// [`MockCollector::enter`], [`MockCollector::exit`], and
+    /// [`MockColletor::drop_span`].
+    ///
+    /// This is especially useful when `tracing-mock` is being used to
+    /// test the traces being generated within your own crate, in which
+    /// case you may need to distinguish between spans which have
+    /// identical metadata but different field values, which can only
+    /// be checked in [`MockCollector::new_span`].
```

Comment:
this is the current state without the PR, right?

Reply:
I meant that the field values can only be checked in new_span. I've added an otherwise in which I think helps clear the confusion (emphasis only for this comment):

> This is especially useful when `tracing-mock` is being used to test the traces being generated within your own crate, in which case you may need to distinguish between spans which have identical metadata but different field values, which can **otherwise** only be checked in [`MockCollector::new_span`].